src/operator/nn/mkldnn/mkldnn_act.cc

```diff
@@ -57,30 +57,56 @@ bool SupportMKLDNNAct(const ActivationParam& param, const NDArray &input) {
   return SupportMKLDNNAct(param);
 }
 
+bool SupportMKLDNNLeakyRelu(const LeakyReLUParam& param) {
+  return param.act_type == leakyrelu::kLeakyReLU
+      || param.act_type == leakyrelu::kELU;
+}
+
+bool SupportMKLDNNLeakyRelu(const LeakyReLUParam& param, const NDArray &input) {
+  // MKL-DNN Activation supports 1d, 2d, 3d, 4d data layout
+  if ((input.shape().ndim() < 1) ||
+      (input.shape().ndim() > 4) ||
+      (input.dtype() != mshadow::kFloat32))
+    return false;
+  return SupportMKLDNNLeakyRelu(param);
+}
+
 bool SupportQuantizedMKLDNNAct(const ActivationParam &param) {
   // TODO(zhennan): Add more activation type when mkldnn supports.
   //                Remove this when it's identity to SupportMKLDNNAct.
   return param.act_type == activation::kReLU;
 }
 
-mkldnn::algorithm GetMKLDNNActAlgo(const ActivationParam& param) {
-  switch (param.act_type) {
-    case activation::kReLU:
-      return mkldnn::algorithm::eltwise_relu;
-    case activation::kSigmoid:
-      return mkldnn::algorithm::eltwise_logistic;
-    case activation::kTanh:
-      return mkldnn::algorithm::eltwise_tanh;
-    case activation::kSoftReLU:
-      return mkldnn::algorithm::eltwise_soft_relu;
-    default:
-      LOG(FATAL) << "unknown activation type";
-      return mkldnn::algorithm::eltwise_relu;
+mkldnn::algorithm GetMKLDNNActAlgo(const MKLDNNActParam& param) {
+  if (param.is_leakyrelu) {
```

Comment:
Why we need to collect `is_leakyrelu` in  `MKLDNNActParam`? I don't see any special for them.

Reply:
addressed